lib/puppet/ssl/host.rb

```diff
@@ -394,6 +394,37 @@ def state
 
   private
 
+  # Load a previously generated CSR either from memory or from disk
+  # @return [Puppet::SSL::CertificateRequest, nil]
+  def load_certificate_request
+    if Puppet::SSL::CertificateRequest.indirection.terminus_class == :memory
+      return Puppet::SSL::CertificateRequest.indirection.find(cert_name)
+    end
+
+    request_path = certificate_request_location(name)
+    if Puppet::FileSystem.exist?(request_path)
+      Puppet::SSL::CertificateRequest.from_s(Puppet::FileSystem.read(request_path))
+    end
+  end
+
+  # Submit the CSR to the CA, either via an HTTP PUT request, or when testing,
+  # via the indirector. This also caches a copy of the CSR on disk.
+  # @param [Puppet::SSL::CertificateRequest] csr the request to submit
+  def submit_certificate_request(csr)
+    if Puppet::SSL::CertificateRequest.indirection.terminus_class == :memory ||
+      Puppet::SSL::CertificateRequest.indirection.terminus_class == :ca
+      return Puppet::SSL::CertificateRequest.indirection.save(csr)
```

Comment:
is there an expected return value from this function?

Reply:
Looks like save does return _something_ (of course we don't well document what), but I don't think `submit_certificate_request` ought to, so I'll move the return.

Reply:
or just return nil